src/main/java/com/internet/shop/util/HashUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.internet.shop.util;
+
+import com.internet.shop.model.User;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+
+    public static byte[] getSalt() {
+        SecureRandom random = new SecureRandom();
+        byte[] salt = new byte[16];
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-512");
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+        return hashedPassword.toString();
+    }
+
+    private boolean isValid(String password, byte[] salt) {
+        User user = new User();
+        user.setPassword(hashPassword(password, salt));
+        user.setSalt(salt);
+
+        return hashPassword(password,salt).equals(user.getPassword());
+    }
```

Comment:
why do you need this method if you don't use it? And do you sure that u need logic for hashing in check method?